From 3dbced2df2942fd8ec4dffead8345a9b9ce7faf0 Mon Sep 17 00:00:00 2001
From: Nicholas Corgan <nick.corgan@ettus.com>
Date: Wed, 5 Aug 2015 09:58:53 -0700
Subject: [PATCH 10/10] Fixed .ipp include behavior

* Use full include path in public headers, not quotes
* Install .ipp files alongside corresponding .hpp
---
 host/include/uhd/transport/nirio/nirio_fifo.h | 4 ++--
 host/include/uhd/utils/CMakeLists.txt         | 4 +++-
 host/include/uhd/utils/math.hpp               | 6 +++---
 3 files changed, 8 insertions(+), 6 deletions(-)

diff --git a/host/include/uhd/transport/nirio/nirio_fifo.h b/host/include/uhd/transport/nirio/nirio_fifo.h
index c424275..cd47147 100644
--- a/host/include/uhd/transport/nirio/nirio_fifo.h
+++ b/host/include/uhd/transport/nirio/nirio_fifo.h
@@ -1,5 +1,5 @@
 //
-// Copyright 2013-2014 Ettus Research LLC
+// Copyright 2013-2015 Ettus Research LLC
 //
 // This program is free software: you can redistribute it and/or modify
 // it under the terms of the GNU General Public License as published by
@@ -114,7 +114,7 @@ private:    //Members
     static const uint32_t FIFO_LOCK_TIMEOUT_IN_MS = 5000;
 };
 
-#include "nirio_fifo.ipp"
+#include <uhd/transport/nirio/nirio_fifo.ipp>
 
 }}
 
diff --git a/host/include/uhd/utils/CMakeLists.txt b/host/include/uhd/utils/CMakeLists.txt
index 05f6892..6b3d5a5 100644
--- a/host/include/uhd/utils/CMakeLists.txt
+++ b/host/include/uhd/utils/CMakeLists.txt
@@ -1,5 +1,5 @@
 #
-# Copyright 2010-2012 Ettus Research LLC
+# Copyright 2010-2012,2015 Ettus Research LLC
 #
 # This program is free software: you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
@@ -24,6 +24,8 @@ UHD_INSTALL(FILES
     byteswap.ipp
     cast.hpp
     csv.hpp
+    fp_compare_delta.ipp
+    fp_compare_epsilon.ipp
     gain_group.hpp
     log.hpp
     math.hpp
diff --git a/host/include/uhd/utils/math.hpp b/host/include/uhd/utils/math.hpp
index 4f88494..2937ba8 100644
--- a/host/include/uhd/utils/math.hpp
+++ b/host/include/uhd/utils/math.hpp
@@ -1,5 +1,5 @@
 //
-// Copyright 2014 Ettus Research LLC
+// Copyright 2014-2015 Ettus Research LLC
 //
 // This program is free software: you can redistribute it and/or modify
 // it under the terms of the GNU General Public License as published by
@@ -240,7 +240,7 @@ namespace fp_compare {
 } // namespace math
 } // namespace uhd
 
-#include "fp_compare_epsilon.ipp"
-#include "fp_compare_delta.ipp"
+#include <uhd/utils/fp_compare_epsilon.ipp>
+#include <uhd/utils/fp_compare_delta.ipp>
 
 #endif /* INCLUDED_UHD_UTILS_MATH_HPP */
-- 
2.1.4

